src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PrisonerService.kt

```diff
@@ -54,7 +59,36 @@ class PrisonerService(
   private fun exactMatchFound(prisonerMatcher: PrisonerMatcher, personEntity: PersonEntity, person: Person) {
     val matchingPrisoner = prisonerMatcher.getMatchingItem()
     logAndTrackEvent(EXACT_MATCH_MESSAGE, TelemetryEventType.NOMIS_MATCH_FOUND, personEntity, person)
-    personRecordService.addPrisonerToPerson(personEntity, matchingPrisoner)
+    val prisonerDetails = getPrisonerDetails(matchingPrisoner.prisonerNumber)
+    if (prisonerDetails != null) {
```

Comment:
this bit might as well be in getPrisonDetails. would save an `if` statement

Reply:
Will update in the next PR.